sklearn/base.py

```diff
@@ -292,6 +293,20 @@ def __repr__(self):
         return '%s(%s)' % (class_name, _pprint(self.get_params(deep=False),
                                                offset=len(class_name),),)
 
+    def __getstate__(self):
+        return dict(self.__dict__.items(), __version__=__version__)
+
+    def __setstate__(self, state):
+        pickle_version = state.get("__version__", "pre-0.18")
+        if pickle_version != __version__:
+            warnings.warn(
+                "Trying to unpickle estimator from version {} when with using "
```

Comment:
Given the lack of traceback with warnings, we should probably mention the estimator. (Although we could use stacklevel, the previous stack entries will be in the pickle module.) The annoyance of that will be that the message will be printed for every estimator loaded.